test/lib/cli/commands/build.js

```diff
@@ -0,0 +1,87 @@
+const {test} = require("ava");
+const sinon = require("sinon");
+const build = require("../../../../lib/cli/commands/build");
+const normalizer = require("@ui5/project").normalizer;
+const builder = require("@ui5/builder").builder;
+const logger = require("@ui5/logger");
+let normalizerStub = null;
+let builderStub = null;
+let loggerStub = null;
+
+const args = {
+	_: [],
+	dest: "./dist",
+	loglevel: "info",
+	t8r: "npm",
+	translator: "npm"
+};
+const defaultBuilderArgs = {
+	tree: {
+		metadata: {
+			name: "Sample"
+		}
+	},
+	destPath: "./dist",
+	buildDependencies: undefined,
+	dev: false,
+	selfContained: false,
+	devExcludeProject: undefined,
+	includedTasks: undefined,
+	excludedTasks: undefined
+};
+
+test.before("Mocking before test execution", (t) => {
+	normalizerStub = sinon.stub(normalizer, "generateProjectTree");
+	builderStub = sinon.stub(builder, "build").returns(Promise.resolve());
+	loggerStub = sinon.stub(logger, "setShowProgress");
+});
+
+test.after("Restore mocks after test exection", (t) => {
+	loggerStub.restore();
+	builderStub.restore();
+	normalizerStub.restore();
+});
+
+test("ui5 build (default) ", async (t) => {
+	normalizerStub.resolves({
+		metadata:
+		{
+			name: "Sample"
+		}
+	});
+	args._ = ["build"];
+	await build.handler(args);
+	t.deepEqual(builderStub.getCall(0).args[0], defaultBuilderArgs, "default build triggered with expected arguments");
+});
+
+test("ui5 build dev", async (t) => {
+	normalizerStub.resolves({
+		metadata:
+		{
+			name: "Sample"
+		}
+	});
+	args._ = ["build", "dev"];
+	await build.handler(args);
+	t.deepEqual(
+		builderStub.getCall(0).args[0],
+		Object.assign({}, {dev: true}, defaultBuilderArgs),
```

Comment:
Doesn't this still result into `dev: false` as the defaultBuilderArgs are passed as last arg?
Assignment is from right to left.

It seems that the fact that `args` and the stubs are shared across tests and they are executed in parallel, this seems to somehow work.

Reply:
Done with 30dd0b7a60bc7bb418b5ef710969dcc6e6284e7c